ros2topic/ros2topic/api/__init__.py

```diff
@@ -37,6 +38,26 @@ def unsigned_int(string):
     return value
 
 
+def nonnegative_int(inval):
+    ret = int(inval)
+    if ret < 0:
+        # The error message here gets completely swallowed by argparse
+        raise ValueError('Value must be positive or zero')
+    return ret
+
+
+def positive_float(inval):
+    ret = float(inval)
+    if ret <= 0.0:
+        # The error message here gets completely swallowed by argparse
+        raise ValueError('Value must be positive')
+    return ret
+
+
+def duration_ns(inval):
+    return Duration(nanoseconds=positive_float(inval))
```

Comment:
why a float?
this should be an integer, https://github.com/ros2/rclpy/blob/388002a9f235ef5ede0fb76b9fb751d6af0b305f/rclpy/rclpy/duration.py#L23.

Reply:
True! do you think that nanoseconds is a good choice? should I go for seconds instead

Reply:
I'm indifferent about nanoseconds vs seconds. Maybe the originating of the feature request has a preference (@matthews-jca any thoughts?).